storage/src/main/java/tech/pegasys/teku/storage/server/ChainStorage.java

```diff
@@ -222,17 +220,12 @@ public SafeFuture<List<BlobSidecar>> getBlobSidecarsBySlotAndBlockRoot(
       final SlotAndBlockRoot slotAndBlockRoot) {
     return SafeFuture.of(
         () -> {
-          try (final Stream<SlotAndBlockRootAndBlobIndex> keyStream =
-              database.streamBlobSidecarKeys(
-                  slotAndBlockRoot.getSlot(), slotAndBlockRoot.getSlot())) {
-            final List<BlobSidecar> blobSidecars = new ArrayList<>();
-            for (Iterator<SlotAndBlockRootAndBlobIndex> iterator = keyStream.iterator();
-                iterator.hasNext(); ) {
-              final SlotAndBlockRootAndBlobIndex key = iterator.next();
-              blobSidecars.add(database.getBlobSidecar(key).orElseThrow());
-            }
-            return blobSidecars;
-          }
+          final List<SlotAndBlockRootAndBlobIndex> blobSidecarKeys =
```

Comment:
why don't we have directly a getBlobSidecars(SlotAndBlockRoot) method that directly streams the matching blobs?